### PR TITLE
Allow LoggerTests to run multiple times

### LINKED ISSUE #82107: LoggersTests incompatible with tests.iters
**Repro line**:
```
./gradlew ':server:test' --tests "org.elasticsearch.common.logging.LoggersTests" -Dtests.iters=2
```

**Reproduces locally?**:
yes

**Applicable branches**:
master, more?

### PR DESCRIPTION
**Pull request by pgomulka**
Log4j configuration do not allow adding the same appender to the same logger multiple times.
LogManager.getLogger(LoggerTests.class) is returning the same logger all the time (caches).
In order to allow adding a new appender instance in the same jvm, different names have to be used.

This fix adds a random number to an appender name to allow to the test run multiple times with
-Dtests.iters

closes #82107

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine update branch

File: server/src/test/java/org/elasticsearch/common/logging/LoggersTests.java [hunk 1]
```diff
@@ -25,7 +26,8 @@
 public class LoggersTests extends ESTestCase {
 
     public void testParameterizedMessageLambda() throws Exception {
-        final MockAppender appender = new MockAppender("trace_appender");
+        // adding a random id to allow test to run multiple times. See AbstractConfiguration#addAppender
+        final MockAppender appender = new MockAppender("trace_appender" + UUID.randomUUID());
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/common/logging/LoggersTests.java:30]: I don’t think a uuid is necessary. It also would not reproduce. Instead you can use a random string or number using the seeded random of the test?

Comment (pgomulka) [reply server/src/test/java/org/elasticsearch/common/logging/LoggersTests.java:30]: good point. will do

**Review by nik9000 — APPROVED** [review]
Thanks!

**Pull request merged**